### ISSUE TITLE
AutoModel extra trial at end of training? "Oracle triggered exit"

### ISSUE THREAD

**Issue by Pythonatheart** [keras-team/autokeras#1064]
I tried to fit AutoModel over image data for classification (3 classes):

input_node = ak.ImageInput()
output_node = ak.ImageBlock(normalize=True, augment=True)(input_node)
output_node = ak.ClassificationHead()(output_node)

auto_model = ak.AutoModel(
    tuner='bayesian',  # tried this because 'greedy' has some issues
    inputs=input_node,
    outputs=output_node,
    max_trials=10,  # maximum models to test is 10?
    overwrite=True)   

auto_model.fit(x_train, y_train, epochs=50)

Training went for 10 trials (10 models) but at the end of the last model (10th trial), this happened:
INFO:tensorflow:Oracle triggered exit

Then, training continues for one more trial (11th trial?), which usually gives the best val_loss? I am not sure is this a bug or does AutoModel do another final training with best model? I have tried max_trials=1 and this still happens i.e. Oracle triggers exit and one final trial happens (total 2 trials).

**Comment by Pythonatheart** [comment]
Just noticed that final_fit is included in the fit method as per discussion in other issues. 
It will be good to update documentation to reflect this?
